pkg/controllers/uiplugin/monitoring.go

```diff
@@ -0,0 +1,125 @@
+package uiplugin
+
+import (
+	"fmt"
+	"strings"
+
+	osv1 "github.com/openshift/api/console/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	uiv1alpha1 "github.com/rhobs/observability-operator/pkg/apis/uiplugin/v1alpha1"
+)
+
+func createMonitoringPluginInfo(plugin *uiv1alpha1.UIPlugin, namespace, name, image string, features []string) (*UIPluginInfo, error) {
+	config := plugin.Spec.Monitoring
+	if config == nil {
+		return nil, fmt.Errorf("monitoring configuration can not be empty for plugin type %s", plugin.Spec.Type)
+	}
+
+	if config.Alertmanager.Name == "" || config.Alertmanager.Namespace == "" {
+		return nil, fmt.Errorf("alertmanager location can not be empty for plugin type %s", plugin.Spec.Type)
+	}
+	if config.ThanosQuerier.Name == "" || config.ThanosQuerier.Namespace == "" {
+		return nil, fmt.Errorf("ThanosQuerier location can not be empty for plugin type %s", plugin.Spec.Type)
+	}
+
+	pluginInfo := &UIPluginInfo{
+		Image:       image,
+		Name:        name,
+		ConsoleName: "acm-monitoring-plugin",
```

Comment:
Although ACM is our first customer for the monitoring plugin, this is not specific to ACM. For example, customers installing the plugin for getting support for Perses dashboards or incidents will load the plugin into the cluster along with the monitoring-plugin provided by CMO. In this case we need to only enable the specific features needed to avoid duplicating the views. Maybe this could be:

```suggestion
		ConsoleName: "monitoring-console-plugin",
```

Reply:
I addition to that we could do a check if the `monitoring-plugin` is present on the cluster to clean up the features to avoid duplication. WDYT?

Reply:
1) The ConsoleName has been changed to "monitoring-console-plugin" and the display name has been changed to "Monitoring Console Plugin".
2) As per our discussion, this deployment will not include the typical monitoring plugin behavior, only the "acm-alerting" specific ones. Checks to enable extra features if they are missing will be left to future work.